### ISSUE TITLE
nextval('hibernate_sequence') does not work in import.sql scripts with H2

### ISSUE THREAD

**Issue by mjurc** [quarkusio/quarkus#15836]
## Describe the bug
Using `nextval('hibernate_sequence')` in `import.sql` files does not work anymore with H2 on Quarkus main, eg the following SQL command

```
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Apple');
```

produces exception with the following root cause:

```
Caused by: org.h2.jdbc.JdbcSQLException: Sequence "HIBERNATE_SEQUENCE" not found; SQL statement:
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Apple') -- (NEXTVAL('hibernate_sequence'), 'Apple') [90036-197]
```

### Expected behavior
`nextval('hibernate_sequence')` works properly as it does in previous versions.

### Actual behavior
`nextval('hibernate_sequence')` does not work.

## To Reproduce

1. Checkout https://github.com/mjurc/quarkus-openshift-test-suite/tree/multiple-persistence-units
2. Run the reproducer test with snapshot build from Quarkus main to reproduce the issue:
   ```
   mvn clean test -pl app-metadata/deployment,app-metadata/runtime,common,sql-db/multiple-pus -Dquarkus-core-only -Dts.retain-on-failure -Dversion.quarkus=999-SNAPSHOT
   ```
3. Run the reproducer test with Quarkus 1.12.1, the issue is not present:
   ```
   mvn clean test -pl app-metadata/deployment,app-metadata/runtime,common,sql-db/multiple-pus -Dquarkus-core-only -Dts.retain-on-failure -Dversion.quarkus=1.12.2.Final
   ```

### Configuration
```properties
%test.quarkus.datasource."fruits".db-kind=h2
%test.quarkus.datasource."fruits".jdbc.url=jdbc:h2:mem:fruits
%test.quarkus.datasource."fruits".username=sa
%test.quarkus.datasource."fruits".password=sa
%test.quarkus.hibernate-orm."fruits".database.charset=utf-8
%test.quarkus.hibernate-orm."fruits".database.generation=drop-and-create
%test.quarkus.hibernate-orm."fruits".sql-load-script=import-fruits.sql
%test.quarkus.hibernate-orm."fruits".datasource=fruits
%test.quarkus.hibernate-orm."fruits".packages=io.quarkus.ts.openshift.sqldb.multiplepus.model.fruit

%test.quarkus.datasource."vegetables".db-kind=h2
%test.quarkus.datasource."vegetables".jdbc.url=jdbc:h2:mem:vegetables
%test.quarkus.datasource."vegetables".username=sa
%test.quarkus.datasource."vegetables".password=sa
%test.quarkus.hibernate-orm."vegetables".database.charset=utf-8
%test.quarkus.hibernate-orm."vegetables".database.generation=drop-and-create
%test.quarkus.hibernate-orm."vegetables".sql-load-script=import-vegetables.sql
%test.quarkus.hibernate-orm."vegetables".datasource=vegetables
%test.quarkus.hibernate-orm."vegetables".packages=io.quarkus.ts.openshift.sqldb.multiplepus.model.vegetable
```
`import-fruits.sql`:
```
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Apple');
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Pear');
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Strawberry');
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Plum');
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Cherry');
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Berry');
INSERT INTO fruit (id, name) VALUES (nextval('hibernate_sequence'), 'Cranberry');
```
`import-vegetables.sql`:
```
INSERT INTO vegetable (id, name) VALUES (nextval('hibernate_sequence'), 'Potato');
INSERT INTO vegetable (id, name) VALUES (nextval('hibernate_sequence'), 'Carrot');
INSERT INTO vegetable (id, name) VALUES (nextval('hibernate_sequence'), 'Parsley');
INSERT INTO vegetable (id, name) VALUES (nextval('hibernate_sequence'), 'Kohlrabi');
INSERT INTO vegetable (id, name) VALUES (nextval('hibernate_sequence'), 'Leek');
INSERT INTO vegetable (id, name) VALUES (nextval('hibernate_sequence'), 'Onion');
INSERT INTO vegetable (id, name) VALUES (nextval('hibernate_sequence'), 'Garlic');
```

### Screenshots
N/A

### Output of `uname -a` or `ver`:
```
Linux tigris 5.10.22-200.fc33.x86_64 #1 SMP Tue Mar 9 22:05:08 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
### Output of `java -version`: 
```
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment 18.9 (build 11.0.10+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.10+9, mixed mode, sharing)
```
### Quarkus version or git rev
```
4344a95c839f80c3ce4d1ef8433beacb4400eb90
```
### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
```
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T21:00:29+02:00)
Maven home: /home/mjurc/.m2/wrapper/dists/apache-maven-3.6.1-bin/38pn40mp89t5c94bjdbeod370m/apache-maven-3.6.1
Java version: 11.0.10, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-11-openjdk-11.0.10.0.9-0.fc33.x86_64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.10.22-200.fc33.x86_64", arch: "amd64", family: "unix"
```

## Additional context
N/A

**Comment by yrodiere** [comment]
I think there's been talks to have one sequence per type instead of a global one. That could explain your problem, in theory. I'd be surprised if we changed that all of a sudden, though. I thought that was planned for ORM 6.

@Sanne did we change the way ORM assigns sequences by default, to assign a specific sequence to each type?

**Comment by Sanne** [comment]
@yrodiere no we didn't make any change related to that discussion yet.

N.B. the URL is missing `DB_CLOSE_DELAY=-1`, that's quite tricky to handle when using H2 in memory - so it might relate with the different connection release strategies.

I haven't tried this though.

**Comment by Sanne** [comment]
See also: #15298

**Comment by yrodiere** [comment]
I've just had a closer look at the reproducer, and it turns out this was simply a configuration mistake.

The test was failing simply because we were telling Quarkus to use a MariaDB dialect, while using an H2 database.
I suppose it used to work with Quarkus 1.x because for some reason the dialect was being ignored, or because the H2 dialect was close enough to the MariaDB dialect. In any case, that was fixed in Quarkus 2.0, so now the invalid dialect configuration is a problem and must be fixed.

See https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/321

Closing as invalid; feel free to reopen or to open another ticket if you disagree.
